### PR TITLE
Relocating favicon.ico

### LINKED ISSUE #30: Favicon not displaying properly
The old Feeling Responsive favicon.ico file still seems to be displayed, even though `assets/img/favicon.ico` is the Software Carpentry favicon.

### PR DESCRIPTION
**Pull request by gvwilson**
Does not appear to fix #30.

### CODE REVIEW

**Comment by pbanaszkiewicz** [comment]
@gvwilson: an excellent cheat sheet by Audrey Roy-Greenfeld: https://github.com/audreyr/favicon-cheat-sheet; it appears that you didn't change `<link rel="icon">` while moving the `favicon.ico` file.

**Comment by gvwilson** [comment]
Line 35 of http://dev.software-carpentry.org/index.html is:

```
<link rel="icon" sizes="32x32" 
href="http://dev.software-carpentry.org/assets/img/favicon.ico 
<view-source:http://dev.software-carpentry.org/assets/img/favicon.ico>" />
```

When I open `http://dev.software-carpentry.org/assets/img/favicon.ico`, 
I see the correct (Software Carpentry) icon. Line 41 also has `<link 
rel="icon">`, and points at 
http://dev.software-carpentry.org/assets/img/touch-icon-192x192.png 
view-source:http://dev.software-carpentry.org/assets/img/touch-icon-192x192.png, 
which isa large version of the Feeling Responsive icon, but its size is 
192x192, so it shouldn't be picked up by the browser.  What do you think 
the `<link rel="icon">` should point at?

**Comment by pbanaszkiewicz** [comment]
@gvwilson could you try deploying a version that has `<link rel="icon" href="/favicon.ico">`? (Also, the favicon file should be in top-most directory, if I understand Audrey's cheat sheet.)

**Comment by gvwilson** [comment]
This branch now has favicon.ico in the root directory, and the link in 
_includes/favicon points there. When I build and serve on my laptop 
after clearing cache, the old Feeling Responsive favicon still shows 
up.  Can someone please confirm this behavior on their machine?

**Comment by pbanaszkiewicz** [comment]
@gvwilson I built your branch, and when I run dev server locally (against `_config.yml`), in the source of http://127.0.0.1:4000/ I found this:

``` html
<link rel="icon" sizes="32x32" href="http://dev.software-carpentry.org/favicon.ico" />
```

It's not pointing to the local server instance, it's pointing to the dev.swc.org instead.

When I run the dev server locally against `_config_dev.yml`, the proper favicon popped up, but it's because the favicon code is missing in `<head>` and my Firefox just tried loading the default path.

So yeah, the problem is in the file `_includes/favicon` - link to favicon in it should point relative to the beginning of URL (`href="/favicon.ico"`) or absolute to correct server (`href="http://localhost:4000/favicon.ico"` in case of dev server).

Cheers!

**Comment by gvwilson** [comment]
Right: if you run `jekyll serve` you get the site URL from `config.yml`, 
which is wrong.  You have to run `make serve` to run Jekyll with _both_ 
config files at the same time (see line 23 of `Makefile`) when you're 
running locally - `_config_dev.yml` is an add-on to `_config.yml`, not a 
replacement.  Have you tried that?

**Comment by pbanaszkiewicz** [comment]
@gvwilson I have not tried that because GitHub Pages supports Jekyll in version 2.4.0 (https://pages.github.com/versions/), but only Jekyll 3.0 comes with `--config` parameter.

**Comment by pbanaszkiewicz** [comment]
@gvwilson no, wait a second, I was running Jekyll 0.11 (:O). Let me try again with a newer version.

**Comment by pbanaszkiewicz** [comment]
@gvwilson I confirm that the wrong favicon shows up.

**Comment by pbanaszkiewicz** [comment]
@gvwilson the problem is in line https://github.com/swcarpentry/website/pull/57/files#diff-85de1aed64278e4735b47a22e15ae91bR16, but I suggest you to fix it by simply commenting out these lines: https://github.com/gvwilson/website/blob/fixing-favicon/_config.yml#L163-L173

They will prevent generating links to other favicons (and you can see there's a lot of favicons from Feeling Responsive: https://github.com/gvwilson/website/blob/fixing-favicon/_includes/favicon#L14-L77).

**Comment by gvwilson** [comment]
Right, but those other favicons are there to handle devices with other 
OSes that want other sizes because why should anything be easy??

**Comment by pbanaszkiewicz** [comment]
> Right, but those other favicons are there to handle devices with other
> OSes that want other sizes because why should anything be easy??

That's correct. If you update them with SWC logo in various sizes then everything will be fine and dandy.

Again, on Firefox the correct favicon is not displayed because we point to a higher resolution one (https://github.com/swcarpentry/website/pull/57/files#diff-85de1aed64278e4735b47a22e15ae91bR16).

**Comment by todrobbins** [comment]
See #68 for a resolution to this.

**Comment by gvwilson** [comment]
Superseded by #68.
